hal/src/thumbv7em/clock/v2/gclkio.rs

```diff
@@ -267,60 +278,39 @@ impl<G: GclkNum> GclkOutToken<G> {
 // GclkOutSource
 //==============================================================================
 
-/// A [`GclkOut`] is associated with a [`Gclk`]
-pub trait GclkOutSourceMarker: GclkNum + SourceMarker {}
-
-impl<G: GclkNum> GclkOutSourceMarker for G {}
-
 mod private {
-    use super::*;
-    pub trait GclkOutSource: Source {
-        fn enable_gclk_out(&mut self, polarity: bool);
-        fn disable_gclk_out(&mut self);
-    }
-}
+    use super::{Counter, Driver, Enabled, Gclk, GclkId, GclkSourceId, NotGclkInId};
 
-pub(crate) use private::GclkOutSource as PrivateGclkOutSource;
+    type EnabledGclk<T> = Enabled<
```

Comment:
This feels like a _type hack_ and feels for me out of place. Isn't `AsEnabledGclk` redundant if we are matching on `Driver<Source = G>` where `G: GclkId`? Am I missing something?

Reply:
Right, it cannot be `GclkInId`.. I wonder if there's a simplier way.

Reply:
Well, we could skip `Driver` here and directly require `Enabled<Gclk<G, S>, N>`. I believe that was the approach taken by the earliest implementation of `GclkOut`. I think you didn't like it though, because it was inconsistent with the rest of the `clock` module.

We could also go back to your approach that defined redundant `enable_gclk_out` functions on a `private` trait. That works too. I was just kind of experimenting with this approach.

I agree that it feels a little hacky. The least hacky approach is probably `Enabled<Gclk<G, S>, N>`.

Reply:
I think this would compile. Looks pretty clean to me.

```rust
impl<G, I> GclkOut<G, I>
where
    G: GclkId,
    I: GclkIo<G>,
{
    pub fn enable<S, N>(
        token: GclkOutToken<G>,
        pin: impl AnyPin<Id = I>,
        mut gclk: Enabled<Gclk<G, S>, N>,
        polarity: bool,
    ) -> (GclkOut<G, I>, Enabled<Gclk<G, S>, N::Inc>)
    where
        S: NotGclkIn,
        N: Increment,
    {
        let freq = gclk.freq();
        let pin = pin.into().into_alternate();
        gclk.enable_gclk_out(polarity);
        let gclk_out = GclkOut { token, freq, pin };
        (gclk_out, gclk.inc())
    }
}
```

Reply:
Yeah, maybe we should operate directly on `Enabled<Gclk>`. Lesser evil. `GclkOut` has it's quirks so it's probably always going to be awkward to model within our design and conventions.

Reply:
I agree. I would also prefer the abstract, `D: Driver` approach, but I don't think there is any way to do in cleanly. This is probably the cleanest approach. I'll make the switch.